src/game/Spell.cpp

```diff
@@ -4508,6 +4508,14 @@ SpellCastResult Spell::CheckCast(bool strict)
         {
             case SPELL_EFFECT_DUMMY:
             {
+                //By Spell ID
+                if (m_spellInfo->Id == 19938)               // Awaken Lazy Peon
+                {
+                    Unit* target = m_targets.getUnitTarget();
+                    if (!target || !target->HasAura(17743 /*SPELL_PEON_SLEEP*/) || target->GetEntry() != 10556 /*NPC_SLEEPING_PEON*/)
+                        return SPELL_FAILED_BAD_TARGETS;
+                }
+                //By SpellIconID
```

Comment:
I figured someone might want to do something for a spell both based on its ID and its icon, but I guess if that's ever needed they can still change it back to an if instead of if else.